usr/libexec/greenboot/greenboot

```diff
@@ -7,6 +8,54 @@ SCRIPTS_CHECK_PATHS=("/usr/lib/greenboot/check" "/etc/greenboot/check")
 SCRIPTS_GREEN_PATHS=("/usr/lib/greenboot/green.d" "/etc/greenboot/green.d")
 SCRIPTS_RED_PATHS=("/usr/lib/greenboot/red.d" "/etc/greenboot/red.d")
 
+source_configuration_file() {
+    GREENBOOT_CONFIGURATION_FILE=/etc/greenboot/greenboot.conf
+    if test -f "$GREENBOOT_CONFIGURATION_FILE"; then
+	# shellcheck source=/etc/greenboot/greenboot.conf
+        source $GREENBOOT_CONFIGURATION_FILE
+    fi
+}
+
+source_configuration_file
+function is_disabled {
+    HEALTHCHECK=$1
+    echo "${DISABLED_HEALTHCHECKS,,}" | tr " " '\n' | grep -F -q -x "$HEALTHCHECK"
+}
+
+script_runner () {
+  local scripts_dir=$1; shift
+  local mode=$1; shift
+  local start_msg=$1; shift
+  local required_hc_failed=false
+  echo "$start_msg"
+  for script in $(find "$scripts_dir" -name '*.sh' | sort); do
+    if is_disabled "$(basename "$script")"; then
+      echo "'$(basename "$script")' was skipped, as specified in config"
+    else
+      local rc=0
+      systemd-cat -t "$(basename "$script")" bash "$script" || rc=$?
+      if [ $rc -ne 0 ]; then
+        local failure_msg
+        failure_msg="Script '$(basename "$script")' FAILURE (exit code '$rc')"
+        case "$mode" in
+          "relaxed")
+            echo "<2>$failure_msg. Continuing..." >&2
+            ;;
+          "strict")
+            required_hc_failed=true
+            echo "<0>$failure_msg. Continuing..." >&2
+        esac
+      else
+        echo "<6>Script '$(basename "$script")' SUCCESS"
+      fi
+    fi
+  done
+
+  if [[ $required_hc_failed == true ]]; then
+    exit 1
+  fi
+}
+
 script_runner () {
```

Comment:
Here is the other script_runner function, maybe that's why shellcheck is complaining about the non reachable code?

Reply:
Found the second one, I was just oblivious on that. I removed the redundancy